File: ruby/lib/steam/packets/rcon/rcon_goldsrc_response.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2008-2011, Sebastian Staudt

require 'steam/packets/steam_packet'

# This packet class represents a RCON response packet sent by a GoldSrc server
#
# It is used to transport the output of a command from the server to the client
# which requested the command execution.
#
# @author Sebastian Staudt
# @see GoldSrcServer#rcon_exec
class RCONGoldSrcResponse

  include SteamPacket

  # Creates a RCON command response for the given command output
  #
  # @param [String] command_response The output of the command executed on the
  #        server
  def initialize(command_response)
    super RCON_GOLDSRC_RESPONSE_HEADER, command_response
  end

  # Returns the output of the command execution
  #
  # @return [String] The output of the command
  def response
    @content_data.string[0..-3]
  end

end

File: php/lib/steam/packets/rcon/RCONGoldSrcResponse.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2008-2009, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage RCON packets
 */

require_once STEAM_CONDENSER_PATH . 'steam/packets/rcon/RCONPacket.php';

/**
 * @package Steam Condenser (PHP)
 * @subpackage RCON packets
 */
class RCONGoldSrcResponse extends SteamPacket
{
	public function __construct($commandResponse)
	{
		parent::__construct(SteamPacket::RCON_GOLDSRC_RESPONSE_HEADER, $commandResponse);
	}

	public function getResponse()
	{
		return substr($this->contentData->_array(), 0, -2);
	}
}
?>
